Composite/Composite.Cs.Tests/Tests.cs

```diff
@@ -15,6 +15,36 @@ internal class Simple {
             public int Number { get; set; }
         }
 
+        [Fact]
+        public void ToBatchedTest () {
+            var inputSequence = Enumerable.Range(1, 10)
+                .Select(x => new Simple{ Number = x, })
+                .AsLimited(6);
+            var batches = C.ToBatched(10, x=>x.Number,  inputSequence)
+                .Take(2)
+                .ToArray();
+
+            Assert.Equal(1, batches[0][0].Number);
+            Assert.Equal(2, batches[0][1].Number);
+            Assert.Equal(3, batches[0][2].Number);
+            Assert.Equal(4, batches[0][3].Number);
+            Assert.Equal(5, batches[1][0].Number);
+            Assert.Equal(6, batches[1][1].Number);
+        }
```

Comment:
Looks like ToBatchedTest shows that the method AsBatched does not follow the described behavior because in the second batch the weight of elements greater than the batch size. Maybe we need to correct the method toBatched and add a check that current element size not greater than numRemains?